src/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs

```diff
@@ -403,16 +412,48 @@ private static async ValueTask<ImmutableArray<UnifiedSuggestedActionSet>> GetAct
             return actionSets;
         }
 
-        private static CodeActionKind GetCodeActionKindFromSuggestedActionCategoryName(string categoryName)
+        private static CodeActionKind GetCodeActionKindFromSuggestedActionCategoryName(string categoryName, IUnifiedSuggestedAction suggestedAction)
             => categoryName switch
             {
                 UnifiedPredefinedSuggestedActionCategoryNames.CodeFix => CodeActionKind.QuickFix,
-                UnifiedPredefinedSuggestedActionCategoryNames.Refactoring => CodeActionKind.Refactor,
+                UnifiedPredefinedSuggestedActionCategoryNames.Refactoring => GetRefactoringKind(suggestedAction),
                 UnifiedPredefinedSuggestedActionCategoryNames.StyleFix => CodeActionKind.QuickFix,
                 UnifiedPredefinedSuggestedActionCategoryNames.ErrorFix => CodeActionKind.QuickFix,
                 _ => throw ExceptionUtilities.UnexpectedValue(categoryName)
             };
 
+        private static CodeActionKind GetRefactoringKind(IUnifiedSuggestedAction suggestedAction)
+        {
+            if (suggestedAction is not ICodeRefactoringSuggestedAction refactoringAction)
+                return CodeActionKind.Refactor;
+
+            if (refactoringAction.CodeRefactoringProvider is ExtractInterfaceCodeRefactoringProvider
```

Comment:
Wondering if for these we should just have a marker interface - would help do the inline check below without reflection which would be prefered IMHO

Reply:
Thanks for talking me into doing the thing I wanted to do. Add a `Kind` property to CodeRefactoringProvider.